dependencies.yaml

```diff
@@ -302,7 +302,7 @@ dependencies:
           - sphinx-markdown-tables
       - output_types: conda
         packages:
-          - doxygen=1.8.20 # pre-commit hook needs a specific version.
+          - doxygen=1.9.1  # pre-commit hook needs a specific version.
```

Comment:
This comment must have been copy-pasted from cudf since there's no actual pre-commit hook for doxygen in this repo.
```suggestion
          - doxygen=1.9.1
```